tests/functional/tests.py

```diff
@@ -602,6 +602,8 @@ def test_get_bucket_policy(client, log_output):
         policy_name = client.get_bucket_policy(bucket_name)
         if policy_name != Policy.NONE:
             raise ValueError('Policy name is invalid: ' + policy_name)
+    except APINotImplemented:
+        pass
```

Comment:
Mint logging requirement defined a 3rd value for the `status` field, `NA` or not applicable, other than `PASS` and `FAIL`. I think `NA` can be used with a message `API is not implemented yet` in the`alert` field for this case. 

At least this way, the tester will be notified about the condition and see the reason why test did not pass or fail, instead of being completely silent about the test.
So, I suggest to have a `print(log_output.json_report_not_applicable())` method defined and called in the main except area to handle `APINotImplemented` exception. This could be done blindly for all tests without any need to know what APIs have been implemented or not.

Reply:
It is better that we define first what's NA.

For me, it means, the test function decides to avoid to run tests since it doesn't make sense to do that in the current context.

In our case, we return NA because it doesn't make sense to run bucket policy get/set tests because we are in gateway mode.

But for that, we need to pass a parameter indicating that the server is a in gateway mode. Always interpreting an API NOT IMPLEMENTED as a NA is not correct in my point of view.

Reply:
1. `NA` is followed currently for Ignored cases due to `NotImplemented` S3 error.
2. As per our discussion we decided to ignore `NotImplemented` error for any S3 call irrespective of the server.  This means `mint` or functional tests of SDKs/tools  do not expect a parameter to control `NotImplemented`.  In a practical case, ignoring get and set bucket policy tests are good enough.
3. No test is written/ignored specifically for gateway.  If we so, this will lead to maintain states inside tests which will not be clean.  In future S3 gateway could have full support of policy where Azure/GCS gateway may support to certain degree.

Reply:
so @balamurugana, do you recommend any change in this PR ?